src/sync/impl/apple/network_reachability_observer.cpp

```diff
@@ -48,10 +50,17 @@ NetworkReachabilityStatus reachability_status_for_flags(SCNetworkReachabilityFla
 
 } // (anonymous namespace)
 
-NetworkReachabilityObserver::NetworkReachabilityObserver(util::Optional<std::string> hostname,
-                                                         std::function<void (const NetworkReachabilityStatus)> handler)
+NetworkReachabilityObserver& NetworkReachabilityObserver::shared(util::Optional<util::Logger&> logger_ref)
```

Comment:
Doing reachability using hostnames is preferable because it can give much more accurate results.